src/Dapr.Extensions.Configuration/DaprSecretStoreConfigurationProvider.cs

```diff
@@ -69,9 +97,39 @@ public DaprSecretStoreConfigurationProvider(string store, bool normalizeKey, IRe
             this.client = client;
         }
 
-        private static string NormalizeKey(string key)
+        /// <summary>
+        /// Creates a new instance of <see cref="DaprSecretStoreConfigurationProvider"/>.
+        /// </summary>
+        /// <param name="store">Dapr Secre Store name.</param>
+        /// <param name="normalizeKey">Replace "__" and any custom delimiters with the delimiter ":".</param>
+        /// <param name="customDelimiters">A collection of delimiters that will be replaced by ':' in the key of every secret.</param>
+        /// <param name="metadata">A collection of metadata key-value pairs that will be provided to the secret store. The valid metadata keys and values are determined by the type of secret store used.</param>
+        /// <param name="client">Dapr client used to retrieve Secrets</param>
+        public DaprSecretStoreConfigurationProvider(string store, bool normalizeKey, IEnumerable<string> customDelimiters, IReadOnlyDictionary<string, string> metadata, DaprClient client)
+        {
+            ArgumentVerifier.ThrowIfNullOrEmpty(store, nameof(store));
+            ArgumentVerifier.ThrowIfNull(client, nameof(client));
+
+            this.store = store;
+            this.normalizeKey = normalizeKey;
+            this.customDelimiters = customDelimiters;
+            this.metadata = metadata;
+            this.client = client;
+        }
+
+        private string NormalizeKey(string key)
         {
-            return key.Replace("__", ConfigurationPath.KeyDelimiter);
+            key = key.Replace("__", ConfigurationPath.KeyDelimiter);
+
+            if (this.customDelimiters?.Any() == true)
+            {
+                foreach (var keyDelimiter in this.customDelimiters)
+                {
+                    key = key.Replace(keyDelimiter, ConfigurationPath.KeyDelimiter);
+                }
+            }
+
```

Comment:
I think this is mostly right, I have a few suggestions after seeing this code... It might be better if we could make `__` removable rather than something that's **truly** built in. 

Here's what I'd suggest:

Make the property on `DaprSecretStoreConfigurationSource` look like:

```
public IList<string> KeyDelimiters { get; set; } = new List<string>(){ "--", };
```

Turn your parameter into `IEnumerable<string> `keyDelimiters` and in your new overload, you replace `source.KeyDelimiters`. This way we don't actually special case `--`, and it's something users can remove if they want. This seems a little more composable.

Your thoughts?